docs/content/docs/connectors/datastream/pulsar.md

```diff
@@ -387,27 +391,402 @@ You should enable transaction in the Pulsar `borker.conf` file when using these
+        <td>ProducerName</td>
+        <td>The 95th percentile of send latency in the last refresh interval for a specific producer.</td>
+        <td>Gauge</td>
+    </tr>
+    <tr>
+        <td>PulsarSink.producer."ProducerName".sendLatency99Pct</td>
+        <td>ProducerName</td>
+        <td>The 99th percentile of send latency in the last refresh interval for a specific producer.</td>
+        <td>Gauge</td>
+    </tr>
+    <tr>
+        <td>PulsarSink.producer."ProducerName".sendLatency999Pct</td>
+        <td>ProducerName</td>
+        <td>The 99.9th percentile of send latency in the last refresh interval for a specific producer.</td>
+        <td>Gauge</td>
+    </tr>
+  </tbody>
+</table>
+
+{{< hint info >}}
+- `numBytesOut`, `numRecordsOut`, `numRecordsOutErrors` are retrieved from Pulsar client metrics.
+
+- `currentSendTime` tracks the time from when the producer calls `sendAync()` to
+  the time when the message is acknowledged by the broker. This metric is not available in `NONE` delivery guarantee.
+{{< /hint >}}
+
+The Pulsar producer refreshes its stats every 60 seconds by default. And PulsarSink retrieves the Pulsar producer
+stats every 500ms. Thus `numRecordsOut`, `numBytesOut`, `numAcksReceived`, and `numRecordsOutErrors` 
+is updated every 60 seconds. To increase the metrics refresh frequency, you can change
+the Pulsar producer stats refresh interval to a smaller value (minimum 1 second), as shown below.
+
+```java
+builder.setConfig(PulsarOptions.PULSAR_STATS_INTERVAL_SECONDS. 1L)
+```
+
+`numBytesOutRate` and `numRecordsOutRate` are calculated based on the `numBytesOut` and `numRecordsOUt`
+counter respectively. Flink internally uses a fixed 60 seconds window to calculate the rates.
+
+### Brief Design Rationale
+
+Pulsar sink follow the Sink API defined in 
+[FLIP-191](https://cwiki.apache.org/confluence/display/FLINK/FLIP-191%3A+Extend+unified+Sink+interface+to+support+small+file+compaction).
+
+#### Stateless SinkWriter
+
+In `EXACTLY_ONCE` mode, the Pulsar sink does not store transaction information in checkpoint,
+which means a new bunch of transactions will be created after a restart.
+Therefore, any message in previous pending transactions is  either aborted or timeout
+(In no way they are visible to the downstream Pulsar consumer).
+Pulsar team is working to optimize the resource cost bought by unfinished pending transactions.
+
+#### Pulsar Schema Evolution
+
+[Pulsar Schema Evolution](https://pulsar.apache.org/docs/en/schema-evolution-compatibility/) allows
+you to reuse the same Flink job after certain "allowed" data model changes, like adding or deleting
+a field in a AVRO-based Pojo class. Please note that you can specify Pulsar schema validation rules
+and define an auto schema update. For details, refer to [Pulsar Schema Evolution](https://pulsar.apache.org/docs/en/schema-evolution-compatibility/).
+
+## Known Issues
+
+This section describes some known issues about the Pulsar connectors.
```

Comment:
Don't we also still have a known issue with Java 11 support? I would like to have that listed too

Reply:
Added~